src/Symfony/Resources/config/services.yml

```diff
@@ -22,20 +22,23 @@ services:
         arguments:
             $toggle: '@SolidWorx\Toggler\Toggle'
 
-    Symfony\Component\DependencyInjection\ExpressionLanguageProvider:
+    toggler.expression.language.provider.dependency_injection:
+        class: Symfony\Component\DependencyInjection\ExpressionLanguageProvider
         public: false
 
-    Symfony\Component\Security\Core\Authorization\ExpressionLanguageProvider:
+    toggler.expression.language.provider.security:
+        class: Symfony\Component\Security\Core\Authorization\ExpressionLanguageProvider
         public: false
 
-    Symfony\Component\ExpressionLanguage\ExpressionLanguage:
+    toggler.expression.language:
+        class: Symfony\Component\ExpressionLanguage\ExpressionLanguage
         public: false
         calls:
-            - ['registerProvider', ['@Symfony\Component\DependencyInjection\ExpressionLanguageProvider']]
-            - ['registerProvider', ['@Symfony\Component\Security\Core\Authorization\ExpressionLanguageProvider']]
+            - ['registerProvider', ['@toggler.expression.language.provider.dependency_injection']]
+            - ['registerProvider', ['@toggler.expression.language.provider.security']]
 
     SolidWorx\Toggler\Toggle:
-        arguments: ['?', '@Symfony\Component\ExpressionLanguage\ExpressionLanguage']
+        arguments: ['?', '@toggler.expression.language']
 
     SolidWorx\Toggler\Symfony\Toggle:
```

Comment:
@pierredup I noticed that the service container gets set in this Symfony service, and that services are retrieved from the container. Should I update that too by removing the `setContainer`, ad rather injecting the needed services?

Reply:
The container is added as a variable for the expression language, so removing it would remove that little benefit. But also, with most services being private, I also don't think there is a lot of values that can be taken from the container.
Let's leave it in for now, and just give it some more thought on how the container should be handled

Reply:
Thanks for the feedback. Happy to leave it as is then for now